### DISCUSSION TITLE
如何自动初始化 Gitalk/Gitment 评论 · /git-comments-initialize

### DISCUSSION THREAD

**Discussion by draveness** [draveness/blog-comments#509] [Announcements]
https://draveness.me/git-comments-initialize

**Comment by Desgard** [comment]
学习到这个姿势了。

**Comment by draveness** [comment]
@Desgard 还是不是美滋滋

**Comment by yunisSong** [comment]
6666

**Comment by CrazyWr** [comment]
脚本变量上下不一致  new_token  token 
不过倒不是什么大事
谢过脚本

**Comment by draveness** [comment]
> @CrazyWr
> 脚本变量上下不一致  new_token  token 
> 不过倒不是什么大事
> 谢过脚本

我一会改一下吧

**Comment by heroydx** [comment]
请问老师，出现Error: Validation Failed.要怎么修改一下啊。。。

**Comment by qinyuanpei** [comment]
请问这个token是从Github的Setting里获得的Personal Access Token吗？

**Comment by noname007** [comment]
hhaddd

**Comment by zouyaoji** [comment]
![报错](https://ws1.sinaimg.cn/large/7a5fa15egy1fp70cqsh0tj21170b4dhm)

很气啊，我的就报这个错，请问博主知道咋回事吗？

**Comment by taoyouxian** [comment]
在使用ruby脚本时出现“{"message":"Validation Failed"”，请问有什么问题吗？

**Comment by draveness** [comment]
> @taoyouxian
> > @Draveness
> > > @CrazyWr
> > > 脚本变量上下不一致  new_token  token 
> > > 不过倒不是什么大事
> > > 谢过脚本
> > 
> > 我一会改一下吧
> 
> 你好，请问你那边是成功了吗？  我这边报错“Validation Failed”

肯定成功了。。不然我发上来干啥

**Comment by iarchean** [comment]
这个很不错，谢谢。

**Comment by xuepro** [comment]
博主的文章说的不清楚，看来根本不知道怎么说

**Comment by qinyuanpei** [comment]
你们这是有多菜啊，天天说这个问题，首先，在个人设置里创建一个token，记下关键的clientId和secretKey，因为这个信息只会显示一次，然后初始化评论不就是创建一个issue吗？参考Github API v3：[https://developer.github.com/v3/issues/#create-an-issue](https://developer.github.com/v3/issues/#create-an-issue)，你发现这里需要一个title和body，tag这个参数可选，但GitmentG和Gitalk都自己定义了一个相对应的tag，这是为了在加载评论时起到一个过滤条件的作用，clientId和secretKey怎么用文档里都介绍得非常清楚，需要注意的是token要要加在header上，格式是：token xxxxxxxxxxxxx，我曾经尝试过hexo g的时候，自己去初始化，然后将issueNumber记录到博客的fontmatter里，后来发现github对API调用次数做了限制，可能所有静态页面都生成了，这个API都还没有调完，nodejs里还需要处理Gzip的坑……总之，折腾不出来就不要折腾了，你以为会有谁给你评论吗？很多时候，博客都是写给我们自己看的，抱歉，我不想再收到这个邮件了，更不要来问我具体怎么弄，谢谢！

**Comment by bingyuea** [comment]
翻到最后一个评论恍然大悟

**Comment by bingyuea** [comment]
哈哈

**Comment by xumeng** [comment]
> @heroydx
> 请问老师，出现Error: Validation Failed.要怎么修改一下啊。。。
有个办法：改下代码截取url一部分，不超过50个字符即可。
具体可参考http://amonxu.com/2018/05/10/Hello-Gitment/

**Comment by xumeng** [comment]
> @taoyouxian
> 在使用ruby脚本时出现“{"message":"Validation Failed"”，请问有什么问题吗？
有个办法：改下代码截取url一部分，不超过50个字符即可。
具体可参考http://amonxu.com/2018/05/10/Hello-Gitment/

**Comment by xumeng** [comment]
非常感谢博主提供的脚本，帮了大忙。致敬！

**Comment by ripperhe** [comment]
额额额额  可以的

**Comment by xigrug** [comment]
还是不太明白

**Comment by ShixiangWang** [comment]
可以应用到hugo发布的博客吗？

**Comment by ibambooooo** [comment]
xuexi

**Comment by chenjiancan** [comment]
原来如此

**Comment by ly918** [comment]
MARK

**Comment by achuanya** [comment]
登录速度限制了怎么办.... www.achuan.io

**Comment by zexuneu** [comment]
mark,,,

**Comment by JustDoDT** [comment]
不错

**Comment by mortyo** [comment]
迭代

**Comment by GODGQC** [comment]
666

**Comment by GODGQC** [comment]
好厉害

**Comment by draveness** [comment]
> @wjup test

你在干啥？？

**Comment by huris** [comment]
学习了，谢谢

**Comment by muziapple** [comment]
>GitHub 中 issue 的可以创建但是并不能删除

是可以删除的，点进issue有个"Delete issue"

**Comment by draveness** [comment]
这好像是刚加的功能，当时没有

**Comment by FreedomLuffy** [comment]
haha

**Comment by yuanshenjian** [comment]
有点小遗憾，gittalk必需翻墙才能看到

**Comment by draveness** [comment]
> 有点小遗憾，gittalk必需翻墙才能看到

你确定么，我没翻墙也可以看啊

**Comment by weiweitop** [comment]
可以看的，只是头像显不出来

**Comment by draveness** [comment]
> 可以看的，只是头像显不出来

GitHub 的头像服务好像被墙了

**Comment by Qnurye** [comment]
> @draveness
> > 可以看的，只是头像显不出来
> 
> GitHub 的头像服务好像被墙了

被gfw域名污染了，改一下hosts就可以显示出来

**Comment by David-Zhang-test** [comment]
> @draveness
> > 有点小遗憾，gittalk必需翻墙才能看到
> 
> 你确定么，我没翻墙也可以看啊

我好像也不用吧

**Comment by kalicn** [comment]
我没翻墙可以啊，这位博主真行

**Comment by draveness** [comment]
> 我没翻墙可以啊，这位博主真行

这个也不是天天墙，一阵一阵的

**Comment by kalicn** [comment]
有时好像也被墙了，尴尬

**Comment by kalicn** [comment]
raw.githubusercontent.com加载不出来

**Comment by kalicn** [comment]
> @muziapple
> >GitHub 中 issue 的可以创建但是并不能删除
> 
> 是可以删除的，点进issue有个"Delete issue"

没有啊

**Comment by uniquezhangqi** [comment]
不用登录吧

**Comment by Ansen** [comment]
参考博主的用python撸了一个轮子：[利用 Github Action 自动初始化 Gitalk 评论之Python篇](https://www.lshell.com/post/use-github-action-and-python-to-automatically-initialize-gitalk-comments/)

**Comment by suehoo3** [comment]
6666

**Comment by JairusTse** [comment]
学习了

**Comment by k6688516518** [comment]
赫本

**Comment by Derrickqi** [comment]
。。

**Comment by 611de** [comment]
6666

**Comment by amor-open** [comment]
好！
